templates/api.yaml

```diff
@@ -0,0 +1,144 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ .Values.env }}-sdapi
+  namespace: {{ .Values.namespace }}
+  labels:
+    app: api
+    env: {{ .Values.env }}
+    tier: svc
+spec:
+  ports:
+  - port: 80
+    targetPort: 80
+  selector:
+    app: api
+    env: {{ .Values.env }}
+    tier: app
+---
+# requires service account to be created
+apiVersion: extensions/v1beta1
```

Comment:
Should we use this old api version for Deployment?

Reply:
looks like for versions before k8s 1.9, we cannot use `app/v1` :(  Not sure if we should make any requirements or assumptions on what k8s version they need to be on.
https://kubernetes.io/docs/concepts/workloads/controllers/deployment/

Reply:
Thanks for reviewing this huge PR! lol 

Looks like for versions before k8s 1.9, we cannot use `app/v1` :(  Not sure if we should make any requirements or assumptions on what k8s version they need to be on.
https://kubernetes.io/docs/concepts/workloads/controllers/deployment/